src/modules/quizzes.js

```diff
@@ -0,0 +1,210 @@
+/* eslint-disable object-curly-newline, no-underscore-dangle */
+const qs = require('qs');
+const nodeFetch = require('node-fetch').default;
+const helpers = require('../utils/helpers');
+
+// Create URL from supplied quizId and parameters
+function createQuizUrl(quizId, parameters, options, path) {
+  const {
+    apiKey,
+  } = options;
+  const serviceUrl = 'https://quizzes.cnstrc.com';
+  let queryParams = { };
+  let answersParamString = '';
+
+  queryParams.index_key = apiKey;
+
+  // Validate quiz id is provided
+  if (!quizId || typeof quizId !== 'string') {
+    throw new Error('quizId is a required parameter of type string');
+  }
+
+  if (path === 'finalize' && (typeof parameters.a !== 'object' || !Array.isArray(parameters.a) || parameters.a.length === 0)) {
+    throw new Error('a is a required parameter of type array');
+  }
+
+  if (parameters) {
+    const { section, a, versionId } = parameters;
+
+    // Pull section from parameters
+    if (section) {
+      queryParams.section = section;
+    }
+
+    // Pull version_id from parameters
+    if (versionId) {
+      queryParams.version_id = versionId;
+    }
+
+    // Pull a from parameters and transform
+    if (a) {
+      a.forEach((ans) => {
+        answersParamString += `&${qs.stringify({ a: ans }, { arrayFormat: 'comma' })}`;
+      });
+    }
+  }
+
+  queryParams._dt = Date.now();
+  queryParams = helpers.cleanParams(queryParams);
+
+  const queryString = qs.stringify(queryParams, { indices: false });
+
+  return `${serviceUrl}/v1/quizzes/${encodeURIComponent(quizId)}/${encodeURIComponent(path)}/?${queryString}${answersParamString}`;
+}
+
+/**
+ * Interface to quiz related API calls
+ *
+ * @module quizzes
+ * @inner
+ * @returns {object}
+ */
+class Quizzes {
+  constructor(options) {
+    this.options = options || {};
+  }
+
+  /**
+   * Retrieve next quiz from api
+   *
+   * @function getNextQuiz
+   * @description Retrieve next quiz from Constructor.io API
+   * @param {string} id - The id of the quiz
+   * @param {string} [parameters] - Additional parameters to refine result set
+   * @param {string} [parameters.section] - Section for customer's product catalog (optional)
+   * @param {array} [parameters.a] - An array for answers in the format [[1,2],[1]]
+   * @param {string} [parameters.version_id] - Specific version id for the quiz.
+   * @param {object} [userParameters] - Parameters relevant to the user request
```

Comment:
Search and browse had userParameters like below so I added some of them here. I'm not sure if these endpoints will be tracked on a user basis, or if I should remove these parameters.

`   * @param {object} [userParameters] - Parameters relevant to the user request
   * @param {number} [userParameters.sessionId] - Session ID, utilized to personalize results
   * @param {number} [userParameters.clientId] - Client ID, utilized to personalize results
   * @param {string} [userParameters.userId] - User ID, utilized to personalize results
   * @param {string} [userParameters.segments] - User segments
   * @param {string} [userParameters.testCells] - User test cells
   * @param {string} [userParameters.userIp] - Origin user IP, from client
   * @param {string} [userParameters.userAgent] - Origin user agent, from client`